tchannel-core/src/test/java/com/uber/tchannel/handlers/InitDefaultRequestHandlerTest.java

```diff
@@ -51,7 +52,7 @@ public void testInitHandlerRemovesItself() throws Exception {
 
         // Given
         EmbeddedChannel channel = new EmbeddedChannel(
```

Comment:
The new classes are mostly changes of the backend logic. Most of the current test should cover the correct behaviors. I have also updated the init test to validate the headers.

I will add some validations in the code logic to make sure a req can never be sent without a identification; however it will not be in this PR.

I will also add tests for peer/connection maintenance and clean up, but that wouldn't be necessary until the peer/connection reaping logic is in.